framework/src/modules/reward/api.ts

```diff
@@ -12,17 +12,56 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
+import { ImmutableAPIContext, BlockHeader, BlockAssets } from '../../node/state_machine';
 import { BaseAPI } from '../base_api';
-import { BFTAPI, RandomAPI } from './types';
+import { calculateDefaultReward } from './calculate_reward';
+import { REWARD_REDUCTION_FACTOR_BFT } from './constants';
+import { APIInitArgs, BFTAPI, RandomAPI } from './types';
 
 export class RewardAPI extends BaseAPI {
 	private _bftAPI!: BFTAPI;
 	private _randomAPI!: RandomAPI;
+	private _brackets!: ReadonlyArray<bigint>;
+	private _offset!: number;
+	private _distance!: number;
+
+	public init(args: APIInitArgs) {
+		this._brackets = args.config.brackets;
+		this._offset = args.config.offset;
+		this._distance = args.config.distance;
+	}
 
 	public addDependencies(bftAPI: BFTAPI, randomAPI: RandomAPI): void {
 		this._bftAPI = bftAPI;
 		this._randomAPI = randomAPI;
-		// eslint-disable-next-line no-console
-		console.log(this._bftAPI, this._randomAPI);
+	}
+
+	public async getBlockReward(
+		context: ImmutableAPIContext,
+		header: BlockHeader,
+		assets: BlockAssets,
+	): Promise<bigint> {
+		const isValidSeedReveal = await this._randomAPI.isValidSeedReveal(
+			context,
+			header.generatorAddress,
+			assets,
+		);
+		if (!isValidSeedReveal) {
+			return BigInt(0);
+		}
+
+		const defaultReward = calculateDefaultReward({
```

Comment:
Could you create an issue to test this function for
- before/at/after the distance height + offset
I see the only before `offset` for this function

Reply:
👍